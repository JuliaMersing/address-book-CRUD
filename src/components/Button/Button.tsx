import React, { FormEvent } from 'react';

interface ButtonProps {
	children: string;
	onClick?: (e: FormEvent) => void;
}

export const Button: React.FC<ButtonProps> = ({ children, onClick }) => (
	<div>
		<button type="submit" className="button-container" onClick={onClick}>
			{children}
		</button>
	</div>
);
