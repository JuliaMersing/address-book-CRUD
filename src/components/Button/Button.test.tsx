import React from 'react';
import { render, screen } from '@testing-library/react';
import { Button } from './Button';

describe('Button component', () => {
	test('should match snapshot add contact', () => {
		render(<Button>Add Contact</Button>);
		screen.getByRole('button', { name: /Add Contact/i });
		expect('button').toMatchSnapshot();
	});
	test('should match snapshot', () => {
		render(<Button>Update contact</Button>);
		screen.getByRole('button', { name: /Update contact/i });
		expect('button').toMatchSnapshot();
	});
});
