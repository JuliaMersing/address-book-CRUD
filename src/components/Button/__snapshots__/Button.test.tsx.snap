// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `"button"`;

exports[`Button component should match snapshot add contact 1`] = `"button"`;
